src/Controllers/PushController.js

```diff
@@ -58,7 +60,55 @@ export class PushController extends AdaptableController {
     body['expiration_time'] = PushController.getExpirationTime(body);
     // TODO: If the req can pass the checking, we return immediately instead of waiting
     // pushes to be sent. We probably change this behaviour in the future.
-    rest.find(config, auth, '_Installation', where).then(function(response) {
+    let badgeUpdate = Promise.resolve();
+
+    if (body.badge) {
+      var op = {};
+      if (body.badge == "Increment") {
+        op = {'$inc': {'badge': 1}}
+      } else if (Number(body.badge)) {
+        op = {'$set': {'badge': body.badge } }
+      } else {
+        throw "Invalid value for badge, expected number or 'Increment'";
+      }
+      let updateWhere = deepcopy(where);
+
+      // Only on iOS!
+      updateWhere.deviceType = 'ios';
+
+      // TODO: @nlutsenko replace with better thing
+      badgeUpdate = config.database.rawCollection("_Installation").then((coll) => {
+        return coll.update(updateWhere, op, { multi: true });
+      });
+    }
+
+    return badgeUpdate.then(() => {
+      return rest.find(config, auth, '_Installation', where)
+    }).then((response) => {
+      if (body.badge && body.badge == "Increment") {
```

Comment:
This looks fine, what do you think if we would inplace mutate every installation instead before sending?
Which looks like you already sort of do with collection iteration aka `results.reduce`, but this would probably reduce complexity of reading this...